src/control/server/privileges.go

```diff
@@ -66,38 +77,53 @@ func getGID(ext External, usr *user.User, groupName string) (int64, error) {
 						usr.Username, group.Name)
 				}
 			} else {
-				return -1, errors.WithMessage(
+				return "", -1, errors.WithMessage(
 					err, "get group membership")
 			}
 		} else {
 			log.Debugf("group lookup: %+v", err)
 		}
 	}
 
-	gid, err := strconv.ParseInt(_gid, 10, 32)
+	gid, err = strconv.ParseInt(_gid, 10, 32)
 	if err != nil {
-		return -1, errors.WithMessage(err, "parsing gid to int")
+		return "", -1, errors.Wrap(err, "parsing gid to int")
 	}
 
-	return gid, nil
+	return grpName, gid, nil
 }
 
-func chownScmMount(config *configuration, uid int64, gid int64) error {
-	for i, srv := range config.Servers {
-		// chown ScmMount (uid/gid are 32bit)
-		err := config.ext.chown(srv.ScmMount, int(uid), int(gid))
-		if err != nil {
-			return errors.WithMessagef(
-				err,
-				"changing scm mount ownership"+
-					" on I/O server %d", i)
+// chownAll changes ownership of required directories (recursive) and
+// files using user/group derived from config file parameters.
+func chownAll(config *configuration, user string, group string) error {
+	paths := []string{
+		config.SocketDir,
+	}
+
+	for _, srv := range config.Servers {
+		paths = append(paths, srv.ScmMount, srv.LogFile)
+	}
+
+	// add last because we don't want to fail to write before chown proc
+	paths = append(paths, config.ControlLogFile)
```

Comment:
Would it be easier to maintain if we write chown in the code near the code that create the files?

Reply:
could you clarify, do you mean adding the files to a list of "files to chown" at the code where we create the files?

Reply:
If possible, the best way might be

    create file 0, 1, 2
    chown file 0, 1, 2
    ...
    create file 3, 4
    chown file 3, 4
    ...

Maintaining a list "files to chown" might be second best.